memory/allocator.go

```diff
@@ -235,6 +247,76 @@ func (a *ResourceAllocator) allocator() memory.Allocator {
 	return a.Allocator
 }
 
+type GcAllocator struct {
+	*ResourceAllocator
```

Comment:
With the above comment, you can change this to `mem memory.Allocator`.

For this, I would avoid using embedding. Embedding is great for some circumstances, but it automatically exposes whatever you embedded and it would be good to control exactly how the methods are implemented.